### ISSUE TITLE
yolo v8 predict everything else as the one custom class name it was custom trained

### ISSUE THREAD

**Issue by haiwu** [ultralytics/ultralytics#3384]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I followed some tutorial and did custom training using Yolo v8, with custom image set, let's say I downloaded 50 cobra snake images from web, labelled them with bbox, and used them to run 100 epochs for training (used same image set for validation). And then I used the generated 'best.pt' model and tried a few prediction for cobra snake, which worked ok. 

Later I tried to use the same model and try to do prediction on some dog / people images, and it predicts all those as 'cobra', it draws bbox around them and labelled them as 'cobra'. 

Why the custom trained model does not know how to properly label 'dog', 'people'? Is there any way to make it better? I know 50 images is not enough, but it should not go ahead and predict everything as cobra?

**Comment by Laughing-q** [comment]
@haiwu if you want the model to identify dog or person, you need to add these to you dataset and train these images. if you at least do not want your model to incorrectly identify snake, you also need to add these images(dog, person) as background before starting a training.

**Comment by haiwu** [comment]
@Laughing-q I was under the impression that since Yolov8 is pretrained on COCO image set, it would be able to detect those 80 class names from COCO. After reading this one: https://github.com/ultralytics/ultralytics/issues/3066, which matches what you mentioned above. 

Would be helpful if there is an option to allow Yolov8 to train on top of the existing knowledge of COLO image set. 

I am still wondering how to reduce the false positives in terms of object detection later. Say I want to detect cobra snakes only with a camera, but it is very likely there are many other animals (including people) that could be captured by the camera. It would be impossible to iterate all possibilities and provide all those possibilities as image set for YOLOv8 to train. 

How to better address this issue later?

**Comment by glenn-jocher** [comment]
@haiwu, thank you for your question and showing interest in Ultralytics YOLOv8! YOLOv8 indeed provides a pre-trained model on the COCO dataset for use cases that revolve around object detection. 

To reduce false positives in object detection, there are several methods to try, including increasing training set diversity to cover more scenarios, augmenting the training set to increase model robustness, and training on fewer classes to improve detection accuracy on the target object class. Additionally, post-processing techniques, such as non-maximum suppression, can be applied to remove redundant detections and decrease the number of false positives. 

Regarding addressing the issue of only detecting cobra snakes with a camera, you will need a process that can identify the object class, in this case, cobra snakes. Once the class is identified, the YOLOv8 model may be trained specifically on the cobra class, which would help minimize false positives.
